DIPs/DIP1005.md

```diff
@@ -0,0 +1,781 @@
+{
+    return process(value);
+}
+void fun(T)(T value) if (is(T == int)) { ... }
+```
+
+In this situation:
+
+* If `fun` is never used, neither module must be loaded.
+* If `fun` is looked up, it will trigger an unspecified lookup for `condition`. This will trigger loading of both `module_a` and `module_b` (and generally all imports in the current module) so as to look up `condition` and ensure no ambiguity.
+
+The same applies to the setup in which `condition` is imported selectively from `module_a` but `module_b` is entirely imported:
+
+```d
+import module_a : condition;
+import module_b;
+void fun(T)(T value) if (condition!T)
+{
+    return process(value);
+}
+void fun(T)(T value) if (is(T == int)) { ... }
+```
+
+In this case, module `module_b` still needs to be opened if `fun` is looked up to ensure no ambiguity exists for `condition`.
+
+Finally, there is the case when all `import`s specify the list of symbols imported:
+
+```d
+import module_a : condition;
+import module_b : process;
+void fun(T)(T value) if (condition!T)
+{
+    return process(value);
+}
+void fun(T)(T value) if (is(T == int)) { ... }
+```
+
+In this case, fine-grained loading of modules is possible: each module is loaded only if a symbol inside it is used. We refer to this setup as "the selective `import` setup".
+
+To generalize the observations above, fine-grained loading of modules is possible under either (or a combination of) the following circumstances: (a) the `static import` setup; (b) the selective `import` setup.
+
+The advantages of such approaches are:
+
+* Fine-grained loading of imports is achieved with no changes in the language definition, only the implementation.
+* Project discipline may be enforced with relative ease, either manually or by means of simple tools. The rule is: "All private imports must be either `static import` or selective `import`".
+
+The disadvantages are:
+
+* The fine-grained _dependency structure_ is not attained by the selective `import` approach. A declaration using unspecified names does not clarify which `import`s it implicitly relies on. The relationship at distance remains between the `import` and the use thereof.
+* The `static import` setup does not share the issue above, at the cost of being cumbersome to use---all imported symbols must use full lookup everywhere. A reasonable engineering approach would be to define shorter names:
+
+```d
+    static import std.range.primitives;
+    alias isInputRange = std.range.primitives.isInputRange;
+    alias isForwardRange = std.range.primitives.isInputRange;
+    ...
+```
+
+    Such scaffolding is of course undesirable in the first place. Also, at least by the current language rules, such `alias` definitions would need to load the module anyway so as to ensure the name does exist. In order for this idiom to work, it would require subtle changes to the language that specify how certain `alias` declarations are exempt from early checking and delayed to the first actual use.
```

Comment:
http://dlang.org/changelog/2.072.0.html#deferred_alias
Also argumenting with the current implementation in a DIP for a major language change isn't really sound.